client/verta/verta/endpoint/nvidia_gpu.py

```diff
@@ -0,0 +1,85 @@
+from enum import Enum
+
+
+class NvidiaGPU:
+    """
+    Nvidia GPU resources
+
+    The JSON equivalent for this is:
+
+    .. code-block:: json
+
+        {
+            "resources": {"nvidia_gpu": {"number": 1, "model": "V100"}}
+        }
+
+    Parameters
+    ----------
+    number: int > 0
+        Number of GPUs requested to run this endpoint's model.
+    model: :class:`NvidiaGPUModel`, optional
+        Model of GPU requested to run this endpoint's model. Currently available models are
+        specified in the `NvidiaGPUModel` enum.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        from verta.endpoint.resources import Resources, NvidiaGPU, NvidiaGPUModel
+        resources = Resources(nvidia_gpu=NvidiaGPU(1, NvidiaGPUModel.V100)
+
+    """
+
+    NUM_GPUS_ERR_MSG = "`number` must be a number greater than 0"
+    MODEL_ERR_MSG = "`model` must be an instance of `verta.endpoint.NvidiaGPUModel`"
+
+    def __init__(self, number, model=None):
+        self._validate_number_of_gpus(number)
+        if model is not None:
+            self._validate_model(model)
+            # convert to NvidiaGPUModel instance if passed in as str
+            model = NvidiaGPUModel(model)
+        self.number = number
+        self.model = model
+
+    def _validate_number_of_gpus(self, number_of_gpus):
+        if not isinstance(number_of_gpus, int):
+            raise TypeError(self.NUM_GPUS_ERR_MSG)
+        if number_of_gpus <= 0:
+            raise ValueError(self.NUM_GPUS_ERR_MSG)
+
+    def _validate_model(self, model):
+        if isinstance(model, str):
+            model = NvidiaGPUModel(model)
+        if not isinstance(model, NvidiaGPUModel):
+            raise TypeError(self.MODEL_ERR_MSG)
+        return model
```

Comment:
I'd recommend against `return`ing from a validation method. Validation should just do validation, not conversion.